### PR TITLE
[aievec] add aievec.min/max e2e to-llvm tests 

### PR DESCRIPTION
**Pull request by jamestcl-amd**
This PR updates the existing element-wise `aievec.min/max` e2e test to support the to-llvm flow.

Changes:
- Update the conversion pattern of `aievec.min/max` to the llvm intrinsic. The third operand `cmp` is implicitly treated as `sign` to the vmax and vmin intrinsic. Will confirm this to the owner. 
- Add element-wise `aievec.min/max` e2e tests for the to-llvm flow. This includes updating the testbench.cc and the test script. These tests, like other to-cpp tests, go through the simulator to verify the numeric correctness.

Todo:
- `aievec.min/max` e2e test for bf16 datatype fails at the peano clang. Need further investigation on it.

### CODE REVIEW

File: lib/Conversion/AIEVecToLLVM/AIEVecToLLVM.cpp [hunk 1]
```diff
@@ -1584,9 +1584,10 @@ class MaxOpConversion : public mlir::ConvertOpToLLVMPattern<aievec::MaxOp> {
     // create xllvm intrinsic
     Value maxOp = nullptr;
     if (llvm::isa<IntegerType>(resultScaTy)) {
-      // create constant for cmp
+      // create constant for third operand `cmp`
+      // Note: `cmp` is implicitly treated as `sign` to the vmax intrinsic
```
Comment (david-vc) [inline lib/Conversion/AIEVecToLLVM/AIEVecToLLVM.cpp:1588]: Thanks for the comment.
This is an oversight in the definition of aievec.min and aievec.max. The existing testcases assume signed comparison.
These should have an attribute specifying whether it uses a signed comparison or an unsigned comparison.
In the arith dialect, there are 4 min ops: minsi, minui, mininumF, minNumF. We deliberately wanted a single op to handle both the integer and floating point cases, but we failed to distinguish between minsi and minui.
Regarding minimumF vs minNumF, I did not find in the arch doc whether -0 < +0.

**Review by david-vc — APPROVED** [review]
I'm OK to address the missing signed attribute for aievec.min and aievec.max in a later PR.

**Comment by jsetoain** [comment]
Silly question, can't Peano deal with the `arith.min/max` instructions if we just lower them to LLVM IR? It feels like it's the kind of thing that should just work.

**Comment by jamestcl-amd** [comment]
> Silly question, can't Peano deal with the `arith.min/max` instructions if we just lower them to LLVM IR? It feels like it's the kind of thing that should just work.

It works when lowering to generic scalar LLVM, but not with generic vector LLVM (e.g. `@llvm.smax.v32i16`). For the generic vector LLVM, it fails at the legalizer.

**Comment by jsetoain** [comment]
> > Silly question, can't Peano deal with the `arith.min/max` instructions if we just lower them to LLVM IR? It feels like it's the kind of thing that should just work.
> 
> It works when lowering to generic scalar LLVM, but not with generic vector LLVM (e.g. `@llvm.smax.v32i16`). For the generic vector LLVM, it fails at the legalizer.

This should be notified to Peano folks, and they should fix it. Once this works in Peano, we can get rid of this translation 😃

**Pull request merged**